packages/core/src/rules/obsolete_statement.ts

```diff
@@ -287,6 +291,39 @@ CALL TRANSFORMATION OBJECTS: https://help.sap.com/doc/abapdocu_752_index_htm/7.5
           issues.push(issue);
         }
       }
+
+      if (configVersion >= Version.v755 && this.conf.regex) {
+        if (sta instanceof Statements.Find || sta instanceof Statements.Replace) {
+          const concat = staNode.concatTokens().toUpperCase();
+
+          if (concat.includes("REGEX")) {
+            const issue = Issue.atStatement(file, staNode, "REGEX obsolete, use PCRE", this.getMetadata().key, this.conf.severity);
+            issues.push(issue);
+          }
+        }
+        else {
+          const classNameExpression = staNode.findAllExpressions(Expressions.ClassName);
+          const methodNameExpression = staNode.findAllExpressions(Expressions.MethodName);
+
+          if (classNameExpression.length !== 0 && methodNameExpression.length !== 0) {
+            const className = classNameExpression[0].concatTokens();
+            const methodName = methodNameExpression[0].concatTokens();
```

Comment:
Probably slow, but I don't have better idea.
Checking for 0 is fine as it doesn't make much sense to chain these methods.

Reply:
handling deprecated parameters for methods should ideally be handled via abapdoc, but I dont think it has annotations for deprecation?

Reply:
No, unfortunately no annotations for deprecation.
Maybe it should be moved to new rule "obsolete_method" or "obsolete_method_parameter"?